### PR TITLE
feat/PHX-11756-11758 - Impact Api Workflows

### PR DESCRIPTION
**Pull request by TRJ-Implan**
Though there are many changes and files in this commit, this is technically covering the work of PHX-11756, PHX-11757, and PHX-11758

---
Only the Combine Regions related work needs to be reviewed as part of this PR.
- `api/impact/workflows/CombineRegions.md` <- Language-agnostic workflow example
- `api/sampleCode/CSharp/Workflow.sln` <- C# Solution containing examples workflow code
  - For the C# solution, only the Authentication and CombineRegion Workflows are ready for review at this time.
---
All the updates for Run Impact Analysis Workflow
Please review api\impact\workflows\RunImpactAnalysis.md and
api\sampleCode\CSharp\ConsoleApp\Workflows\RunImpactAnalysisWorkflow.cs specifically for changes related to PHX-11758
---

The final updates for the Create Project Workflow have been added:
- `api\impact\workflows\CreateProject.md`
- `api\sampleCode\CSharp\ConsoleApp\Workflows\CreateProjectWorkflow.cs`

At this point all three tickets' workflows should be present in both Markdown form and a corresponding C# example (that should be roughly in the same order and have similar comments)

### CODE REVIEW

File: sampleCode/CSharp/ConsoleApp/Workflows/CreateProjectWorkflow.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        // Get a list of all valid Data Sets for the Aggregation Scheme
+        var datasets = DataSets.GetDataSets(aggregationSchemeId);
+        // Choose the one you would like to use -- must be compatible with your chosen HouseholdSetId
+        int dataSetId = 96;
+
+        
+        // Now we can define the project with its required properties
+        Project project = new Project
+        {
+            Title = $"PHX-11757 - Test - {Guid.NewGuid()}",
```
Comment (devimplan) [inline sampleCode/CSharp/ConsoleApp/Workflows/CreateProjectWorkflow.cs:33]: @TRJ-Implan can we remove the ticket number and test, and replace with "Project Name" then guid?

File: sampleCode/CSharp/ConsoleApp/Services/Rest.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    {
+        // start with a placeholder token (to be overriden with SetAuthentication)
+        _jwtAuthenticator = new JwtAuthenticator("BAD_TOKEN");
+        // setup the defaults for all Rest requests
+        _restClientOptions = new RestClientOptions
+        {
+            // This is the base endpoint for all Implan ImpactAPI Requests
+            BaseUrl = new Uri("https://api.implan.com/int/"),
+            //BaseUrl = new Uri("https://localhost:5001/external/"),
+
```
Comment (MT737) [inline sampleCode/CSharp/ConsoleApp/Services/Rest.cs:28]: Likely still here due to being in WIP status, but just calling out these hard coded environment pathways.

File: sampleCode/CSharp/ConsoleApp/Workflows/CombinedRegionWorkflow.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        RegionalWorkflow.Examples();
+        
+        
+        /* Combine Regions
+         * Once you have found the HashIds of the Regions you wish to combine (using the endpoints above),
+         * you can combine them together into a single Region         
+         */
+
+        // These examples are from Aggregation Scheme 8, DataSet 96
+        string hashId1 = "W1aQl9wzxj";     // Lane County, OR
```
[the remark this answers is not in the document]
Comment (TRJ-Implan) [reply sampleCode/CSharp/ConsoleApp/Workflows/CombinedRegionWorkflow.cs:45]: Yeah, I can expand the Regions examples to include something like that!

**Comment by TRJ-Implan** [comment]
All the updates for Run Impact Analysis Workflow
Please review `api\impact\workflows\RunImpactAnalysis.md` and 
`api\sampleCode\CSharp\ConsoleApp\Workflows\RunImpactAnalysisWorkflow.cs` specifically for changes related to PHX-11758

**Comment by TRJ-Implan** [comment]
The final updates for the Create Project Workflow have been added:
- `api\impact\workflows\CreateProject.md`
- `api\sampleCode\CSharp\ConsoleApp\Workflows\CreateProjectWorkflow.cs`

At this point, all three tickets' workflows should be present in both Markdown form and a corresponding C# example (that should be roughly in the same order and have similar comments)

File: impact/workflows/RunImpactAnalysis.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+### Notes:
+- All API Endpoints require a valid JWT Bearer Token ([JWT.IO](https://jwt.io/))
+  -	Please see the [authentication](https://github.com/Implan-Group/api/blob/main/impact/readme.md#authentication---retrieving-bearer-access-token) section of the primary Readme to review authentication steps
+- Variables for endpoints will appear inside of double-braces, those must be replaced with valid values before execution
+  - e.g. `{{api_domain}}` should be replaced with `https://api.implan.com/`
+  - See the [Development Variables](https://github.com/Implan-Group/api/blob/main/impact/readme.md#development-variables) section of the Readme for more information
+
+---
+# Workflow Overview
```
Comment (devimplan) [inline impact/workflows/RunImpactAnalysis.md:13]: @TRJ-Implan this appears the same as RunImpactAnalysis.md.

File: impact/workflows/CreateProject.md [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+- This document describes the basic IMPLAN ImpactAPI workflow to create a Project, add Events, and link Events and Regions in Groups
+
+### Notes:
+- All API Endpoints require a valid JWT Bearer Token ([JWT.IO](https://jwt.io/))
+  - Please see the [authentication](https://github.com/Implan-Group/api/blob/main/impact/readme.md#authentication---retrieving-bearer-access-token) section of the primary Readme to review authentication steps
+- Variables for endpoints will appear inside of double-braces, those must be replaced with valid values before execution
+  - e.g. `{{api_domain}}` should be replaced with `https://api.implan.com/`
+  - See the [Development Variables](https://github.com/Implan-Group/api/blob/main/impact/readme.md#development-variables) section of the Readme for more information
+- Endpoints that combine IMPLAN's economic data with other data are limited to only those datasets to which those other data are available. (This primarily effects Canada + International customers)
+
```
Comment (devimplan) [inline impact/workflows/CreateProject.md:14]: @TRJ-Implan can we add an overview?

Comment (TRJ-Implan) [reply impact/workflows/CreateProject.md:14]: Are you suggesting a more detailed overview of the Authentication process?

Comment (MT737) [reply impact/workflows/CreateProject.md:14]: I believe this is in reference to having an overview section with a brief explanation of the purpose for a `Project`. Much like you have in your CombinedRegion workflow.

Comment (devimplan) [reply impact/workflows/CreateProject.md:14]: Correct @MT737 .  @TRJ-Implan looking for the same general structure as the other md files.

File: sampleCode/CSharp/ConsoleApp/Workflows/RunImpactAnalysisWorkflow.cs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        
+        // If you want to look at all Projects that you created
+        Project[] projects = Projects.GetProjects();
+        
+        // If you want to look at all Projects that have been shared with you
+        Project[] shared = Projects.GetSharedProjects();
+        
+        // If you want the details for a specific Project you need the Project Id
+        Project project = Projects.GetProject(Guid.Parse("deadbeef-2600-1337-cafe-123456789abc"));
+
```
Comment (MT737) [inline sampleCode/CSharp/ConsoleApp/Workflows/RunImpactAnalysisWorkflow.cs:28]: As opposed to `livebeef`? Fun aside, is the intent to leave a sample GUID?

File: sampleCode/CSharp/ConsoleApp/Workflows/RunImpactAnalysisWorkflow.cs [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+            if (string.Equals(status, "Complete", StringComparison.OrdinalIgnoreCase))
+                break;
+
+            // If it has not yet completed, give it more time to process
+            Thread.Sleep(TimeSpan.FromMinutes(1));
+        }
+        
+        // If the Impact seems to be taking an unusually long time to run or you want to make changes, you can also Cancel a running Impact Analysis
+        bool cancelled = Impacts.CancelImpact(14710);
+
```
Comment (MT737) [inline sampleCode/CSharp/ConsoleApp/Workflows/RunImpactAnalysisWorkflow.cs:52]: Hardcoded value. Use the well named variable for clarity.

**Pull request merged**